pkg/loader/compose/compose.go

```diff
@@ -120,6 +121,28 @@ func checkUnsupportedKey(composeProject *project.Project) []string {
 							yamlTagName = "networks"
 						}
 					}
+
+					if f.Name() == "Links" {
+						//Links has "SERVICE:ALIAS" style, we don't support SERVICE != ALIAS
+						if linksArray := val.FieldByName(f.Name()); linksArray.Kind() == reflect.Slice {
+							findUnsupportedLinksFlag := false
+							for i := 0; i < linksArray.Len(); i++ {
+								if tmpLink := linksArray.Index(i); tmpLink.Kind() == reflect.String {
+									tmpLinkStr := tmpLink.String()
+									tmpLinkStrSplit := strings.Split(tmpLinkStr, ":")
+									if len(tmpLinkStrSplit) == 2 && tmpLinkStrSplit[0] != tmpLinkStrSplit[1] {
+										findUnsupportedLinksFlag = true
+										break
+									}
+								}
+							}
+							if !findUnsupportedLinksFlag {
```

Comment:
Looking at the code, I don't think this is doing anything? Since `findUnsupportedLinksFlag` is outside the `for i := 0; i < linksArray.Len(); i++ {` loop. I could be wrong!

Reply:
This flag `findUnsupportedLinksFlag ` is for the above code `for _, f := range s.Fields() {` at about 100 lines this file.The flag's goal is to find compose links like 
```
  links:
    - 'db:mysql'
```
which service name is different from alias name.   If we find links like this, this flag is set to `true` and code will process, `keysFound = append(keysFound, yamlTagName)` this code will append `links` to `keysFound` and print warning messsage. But if we have links like this
```
  links:                                links:
    - redis                              - 'mysql:mysql'
```
There are many compose like this in our examples and test case folder. For this kind of compose ,we can convert it and shouldn't print any warning message. So in this situation, `findUnsupportedLinksFlag` will set to `false`, and `links` will not be append to `keysFound`.  This code `keysFound = append(keysFound, yamlTagName)` won't run because we have continue, and code will process next filed.

Reply:
Ah, that makes sense. LGTM.